src/NuGet.Core/NuGet.ProjectModel/PackageSpecOperations.cs

```diff
@@ -13,8 +13,11 @@ namespace NuGet.ProjectModel
 {
     public static class PackageSpecOperations
     {
-        public static void AddOrUpdateDependency(PackageSpec spec, PackageDependency dependency)
```

Comment:
I checked that, and it's not used. 
I'd normally prefer to cleanup APIs that aren't super important for other users, but I'll revert this.